secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/dcr/request/DCRSignatureValidationException.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.dcr.request;
+
+import org.forgerock.util.Reject;
+
+final public class DCRSignatureValidationException extends Exception{
```

Comment:
Reuse com.forgerock.sapi.gateway.dcr.ValidationException

Reply:
ValidationException is a runtime exception. I prefer to see the exceptions in the method signatures and be forced to handle them. Better for people following on. This gives a chance to make decent log entries that we know will have the transactionId in them.

When RuntimeExceptions happen and you search logs using tx id, then you don't see the runtime exceptions. You just see that processing stopped. THen you have to pin a log entry, remove the tx id from the search and search again. Then you see the runtime exception. 

This is not good for customers that want to diagnose why a TPPs method call failed.

Reply:
We can make ValidationException a checked exception, the class defs are otherwise identitcal.

I personally prefer RuntimeException which get caught at a suitable level with an Exception Handler i.e. in the Filter.filter method. As you need to think about handling unexpected RuntimeExceptions anyway, so I prefer to not have to pollute all of the sigs with throws ValidationException (and a list of other exceptions as the biz logic complexity increases).

But I'm happy to go down the checked exception path if you prefer.